client/snapctl_test.go

```diff
@@ -82,3 +82,24 @@ func (cs *clientSuite) TestInternalSnapctlCmdNeedsStdin(c *check.C) {
 		c.Check(res, check.Equals, false)
 	}
 }
+
+func (cs *clientSuite) TestClientRunSnapctlReadLimit(c *check.C) {
+	cs.rsp = `{
+		"type": "sync",
+        "status-code": 200,
+		"result": {
+		}
+	}`
+
+	restore := client.MockStdinReadLimit(11)
```

Comment:
Also could use a test where we give it stdinReadLimit bytes in the input and so no error should be returned.

Reply:
Added